pkg/cache/cache_test.go

```diff
@@ -22,40 +22,96 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	kcorev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kcache "k8s.io/client-go/tools/cache"
 
 	"sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const testNamespaceOne = "test-namespace-1"
+const testNamespaceTwo = "test-namespace-2"
+
+func createPod(name, namespace string, restartPolicy kcorev1.RestartPolicy) runtime.Object {
+	three := int64(3)
+	pod := &kcorev1.Pod{
+		ObjectMeta: kmetav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"test-label": name,
+			},
+		},
+		Spec: kcorev1.PodSpec{
+			Containers:            []kcorev1.Container{{Name: "nginx", Image: "nginx"}},
+			RestartPolicy:         restartPolicy,
+			ActiveDeadlineSeconds: &three,
+		},
+	}
+	cl, err := client.New(cfg, client.Options{})
+	Expect(err).NotTo(HaveOccurred())
+	err = cl.Create(context.Background(), pod)
+	Expect(err).NotTo(HaveOccurred())
+	return pod
+}
+
+func deletePod(pod runtime.Object) {
+	cl, err := client.New(cfg, client.Options{})
+	Expect(err).NotTo(HaveOccurred())
+	err = cl.Delete(context.Background(), pod)
+	Expect(err).NotTo(HaveOccurred())
+}
+
 var _ = Describe("Informer Cache", func() {
-	var stop chan struct{}
+
+	var (
+		informerCache cache.Cache
+		stop          chan struct{}
+		knownPod1     runtime.Object
+		knownPod2     runtime.Object
+		knownPod3     runtime.Object
+	)
 
 	BeforeEach(func() {
 		stop = make(chan struct{})
 		Expect(cfg).NotTo(BeNil())
+
+		By("creating three pods")
+		knownPod1 = createPod("test-pod-1", testNamespaceOne, kcorev1.RestartPolicyNever)
```

Comment:
Add a comment that the restartpolicy is used by the field indexing test so folks know not to change these.